crates/compiler/derive_key/src/hash.rs

```diff
@@ -143,9 +143,13 @@ impl FlatHash {
             Content::LambdaSet(_) => Err(Underivable),
         }
     }
+
+    pub fn from_builtin_symbol(symbol: Symbol) -> Result<FlatHash, DeriveError> {
+        builtin_symbol_to_hash_lambda(symbol).ok_or(DeriveError::Underivable)
+    }
 }
 
-const fn num_symbol_to_hash_lambda(symbol: Symbol) -> Option<FlatHash> {
+const fn builtin_symbol_to_hash_lambda(symbol: Symbol) -> Option<FlatHash> {
     use FlatHash::*;
     match symbol {
         Symbol::NUM_U8 | Symbol::NUM_UNSIGNED8 => {
```

Comment:
I guess here we eventually need to add something that casts a `Bool` to a `U8` and then hashes it as a `U8`? Otherwires records with bools in them will not derive hash?

Reply:
Yeah that should work